test/integration/base_integration_test.cc

```diff
@@ -121,11 +122,22 @@ Network::TransportSocketFactoryPtr BaseIntegrationTest::createUpstreamTlsContext
   } else if (upstream_config_.upstream_protocol_ == FakeHttpConnection::Type::HTTP1) {
     tls_context.mutable_common_tls_context()->add_alpn_protocols("http/1.1");
   }
-  auto cfg = std::make_unique<Extensions::TransportSockets::Tls::ServerContextConfigImpl>(
-      tls_context, factory_context_);
-  static Stats::Scope* upstream_stats_store = new Stats::IsolatedStoreImpl();
-  return std::make_unique<Extensions::TransportSockets::Tls::ServerSslSocketFactory>(
-      std::move(cfg), context_manager_, *upstream_stats_store, std::vector<std::string>{});
+  if (upstream_config_.upstream_protocol_ != FakeHttpConnection::Type::HTTP3) {
+    auto cfg = std::make_unique<Extensions::TransportSockets::Tls::ServerContextConfigImpl>(
+        tls_context, factory_context_);
+    static Stats::Scope* upstream_stats_store = new Stats::IsolatedStoreImpl();
+    return std::make_unique<Extensions::TransportSockets::Tls::ServerSslSocketFactory>(
+        std::move(cfg), context_manager_, *upstream_stats_store, std::vector<std::string>{});
+  } else {
+    envoy::extensions::transport_sockets::quic::v3::QuicDownstreamTransport quic_config;
+    quic_config.mutable_downstream_tls_context()->MergeFrom(tls_context);
+
+    std::vector<std::string> server_names;
+    auto& config_factory = Config::Utility::getAndCheckFactoryByName<
+        Server::Configuration::DownstreamTransportSocketConfigFactory>(
+        "envoy.transport_sockets.quic");
```

Comment:
I think we're moving away from wkn files per https://github.com/envoyproxy/envoy/issues/7238
I used hard-coded strings to avoid further use of wkn.  will let Matt comment on his preference on these ones since I do lean towards using predefined values myself.

Reply:
I think it's fine either way for now, but again related to why I think we need to undo all the extension stuff.